### PR TITLE
refactor(api): add common responses for api & prepare for route handlers

### LINKED ISSUE #235: [FEATURE]: better error responses in the API
**Is your feature request related to a problem? Please describe.**
I think that the API responses could be formatted differently, so they all use the same style and provide the necessary info on how to fix the error. Currently, the single `message` field is not enough for proper error handling.

**Describe the solution you'd like**
This is an example I implemented on my own API project, it doesn't have to be just the same but something like this 🙂 
image

Your error response could be: 
```
{
	"status": number
	"message": string
	"code": string // unique for every different error, for e.x: MISSING_ENVIRONMENT_ID, MISSING_SESSION_ID
}
```

**Describe alternatives you've considered**

**Additional context**
With proper error responses, the clients (SDKs and ppl's own projects) could understand the errors more easily and it would improve the code quality by always returning response in a known structure.

### PR DESCRIPTION
**Pull request by kymppi**
Closes #235

### CODE REVIEW

File: apps/web/pages/api/v1/client/environments/[environmentId]/events/index.ts [hunk 1]
```diff
@@ -1,11 +1,12 @@
 import { prisma } from "@formbricks/database";
-import type { NextApiRequest, NextApiResponse } from "next";
+import type { NextApiRequest } from "next";
+import { CustomNextApiResponse, responses } from "../../../../../../../lib/api/response";
```
Comment (mattinannt) [inline apps/web/pages/api/v1/client/environments/[environmentId]/events/index.ts:3]: also here `from "@/lib/api/response` syntax can be used

**Review by mattinannt — COMMENTED** [review]
@kymppi Looks pretty great, good work 💪🎉
I'm a bit conflicting how to continue. Formbricks currently sits between the old nextjs `pages` directory and the new `app` directory. Because of missing compatibility of next-auth with the app-directory / route handlers it wasn't possible to build the api's into the app directory with route handlers ( https://beta.nextjs.org/docs/routing/route-handlers ). I also started a PR where I began to began to move existing api routes to the app directory now next-auth supports it. I will merge this PR tomorrow and that way set the basis for that.
Because we need to change every API route I think changing the API to the app-directory is a good moment to also change the data-handling and responses.
So I think adding the `apps/web/lib/api/response.ts` lib first and then moving the apis step by step (while also making sure it doesn't break functionality) would be a good move; even though it's a lot to do. But that's something I can also do within 1-2 days and 10 cups of coffee ;-)
what do you think?

**Comment by kymppi** [comment]
I think so too, I can remove the api route changes I did and keep the file in `lib/`. Then you could merge this PR without any conflicts. After that the api responses can be changed slowly while making sure the routes still work 👍

**Comment by kymppi** [comment]
Also we can provide full type safety between the api and the api client. We could make interfaces inside the `@formbricks/types` package and use those with the nextjs route handlers and with fetch on the `@formbricks/api` package.

File: apps/web/pages/api/v1/client/environments/[environmentId]/displays/[displayId]/responded.ts [hunk 2]
```diff
@@ -1,17 +1,18 @@
 import { prisma } from "@formbricks/database";
-import type { NextApiRequest, NextApiResponse } from "next";
+import type { NextApiRequest } from "next";
+import { CustomNextApiResponse, responses } from "../../../../../../../../lib/api/response";
```
[the remark this answers is not in the document]
Comment (kymppi) [reply apps/web/pages/api/v1/client/environments/[environmentId]/displays/[displayId]/responded.ts]: Added a setting to the vscode settings, so @/lib/ should work fine with auto imports

**Comment by mattinannt** [comment]
Great job 🎉 Merged as discussed in discord 😊

**Pull request merged**